tests/TestTotp.cpp

```diff
@@ -83,3 +83,58 @@ void TestTotp::testTotpCode()
+        const QTotp::Encoder& enc = iter.value();
+        QVERIFY2(enc.digits != 0,
+            qPrintable(QString("Custom encoders cannot have zero-value for digits field: %1(%2)").arg(enc.name).arg(iter.key())));
+        QVERIFY2(!enc.name.isEmpty(),
+            qPrintable(QString("Custom encoders must have a name: %1(%2)").arg(enc.name).arg(iter.key())));
+        QVERIFY2(!enc.shortName.isEmpty(),
+            qPrintable(QString("Custom encoders must have a shortName: %1(%2)").arg(enc.name).arg(iter.key())));
+        QVERIFY2(QTotp::shortNameToEncoder.contains(enc.shortName),
+            qPrintable(QString("No shortNameToEncoder entry found for custom encoder: %1(%2) %3").arg(enc.name).arg(iter.key()).arg(enc.shortName)));
+        QVERIFY2(QTotp::shortNameToEncoder[enc.shortName] == iter.key(),
+            qPrintable(QString("shortNameToEncoder doesn't reference this custome encoder: %1(%2) %3").arg(enc.name).arg(iter.key()).arg(enc.shortName)));
+        QVERIFY2(QTotp::nameToEncoder.contains(enc.name),
+            qPrintable(QString("No nameToEncoder entry found for custom encoder: %1(%2) %3").arg(enc.name).arg(iter.key()).arg(enc.shortName)));
+        QVERIFY2(QTotp::nameToEncoder[enc.name] == iter.key(),
+            qPrintable(QString("nameToEncoder doesn't reference this custome encoder: %1(%2) %3").arg(enc.name).arg(iter.key()).arg(enc.shortName)));
+    }
+
+    for (QMap<QString, quint8>::const_iterator iter = QTotp::nameToEncoder.cbegin(); iter != QTotp::nameToEncoder.cend(); ++iter) {
+        QVERIFY2(QTotp::encoders.contains(iter.value()),
+                qPrintable(QString("No custom encoder found for encoder named %1(%2)").arg(iter.value()).arg(iter.key())));
+        QVERIFY2(QTotp::encoders[iter.value()].name == iter.key(),
+                qPrintable(QString("nameToEncoder doesn't reference the right custom encoder: %1(%2)").arg(iter.value()).arg(iter.key())));
+    }
+
+    for (QMap<QString, quint8>::const_iterator iter = QTotp::shortNameToEncoder.cbegin(); iter != QTotp::shortNameToEncoder.cend(); ++iter) {
+        QVERIFY2(QTotp::encoders.contains(iter.value()),
+                qPrintable(QString("No custom encoder found for short-name encoder %1(%2)").arg(iter.value()).arg(iter.key())));
+        QVERIFY2(QTotp::encoders[iter.value()].shortName == iter.key(),
+                qPrintable(QString("shortNameToEncoder doesn't reference the right custom encoder: %1(%2)").arg(iter.value()).arg(iter.key())));
+    }
+}
+
+void TestTotp::testSteamTotp()
+{
+    quint8 digits = 0;
+    quint8 step = 0;
+    QString secret = "otpauth://totp/"
+                     "test:test@example.com?secret=63BEDWCQZKTQWPESARIERL5DTTQFCJTK&issuer=Valve&algorithm="
+                     "SHA1&digits=5&period=30&encoder=steam";
+    QCOMPARE(QTotp::parseOtpString(secret, digits, step), QString("63BEDWCQZKTQWPESARIERL5DTTQFCJTK"));
```

Comment:
Are there any official references which say that these values are correct? If yes, please reference them.

Reply:
There are not. I created a throw-away Steam account, then used their Steam app and captured their codes at two separate times. Should I add a note to that effect?

Reply:
Perhaps not a bad idea.